File: js/R13-FACTORIAL-RECURSIVO.js
function factorial(n) {
    //si es negativo, retorna null
    if (n<0) {
        return null //si es negativo no tiene factorial
    }

    if (n<=1) {
        return 1 //si el numero es 1 o 0, es 1
    } else {
        return n * factorial ( n - 1) // para mayor a 1, el factorial se calcula recursivamente
    }

}

function main() {
    console.log(factorial(0) || 'no tiene factorial')
    console.log(factorial(2) || 'no tiene factorial')
    console.log(factorial(1) || 'no tiene factorial')
    console.log(factorial(-1) || 'no tiene factorial')
    console.log(factorial(12) || 'no tiene factorial')
    console.log(factorial(4) || 'no tiene factorial')
}

main()

File: python/R13-FACTORIAL-RECURSIVO.py
def factorial(n):
    #si el numero es negativo, retorna none
    if n<0:
        return None
    #si el numero es 0 o 1, el factorial es 1
    if n <= 1:
        return 1
    else:
        #para numeros mayores a 1, calcula el factorial recursivamente
        return n * factorial(n-1)

def main():
    print(factorial(0) or 'No tiene factorial')
    print(factorial(1) or 'No tiene factorial')
    print(factorial(4) or 'No tiene factorial')
    print(factorial(10) or 'No tiene factorial')
    print(factorial(15) or 'No tiene factorial')
    print(factorial(-10) or 'No tiene factorial')

main()